ntoskrnl/mm/ARM3/section.c

```diff
@@ -1498,6 +1498,8 @@ MiMapViewOfDataSection(IN PCONTROL_AREA ControlArea,
                            AllocationType);
     if (!NT_SUCCESS(Status))
     {
+        MiDereferenceControlArea(ControlArea);
+        ExFreePoolWithTag(Vad, 'ldaV');
```

Comment:
VAD was created after, so it has to disappear first.
```suggestion
        ExFreePoolWithTag(Vad, 'ldaV');
        MiDereferenceControlArea(ControlArea);
```

Reply:
There is also (at least one) modification that has been made on the `Segment`, above:
`Segment->NumberOfCommittedPages += QuotaCharge;`
that probably needs to be rolled back too (to be done under `MmSectionCommitMutex` held).

Reply:
Another (perhaps better?) way would be to insert the VAD before doing the operations on the corresponding Section...